plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/RegisterTableProcedure.java

```diff
@@ -153,9 +155,16 @@ private void doRegisterTable(
         Table table = buildTable(session, schemaTableName, tableLocation, true);
 
         PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(table.getOwner().orElseThrow());
-        // Clear the caches so that createTable verifies we're registering a location with a valid table
         statisticsAccess.invalidateCache(tableLocation);
         transactionLogAccess.invalidateCaches(tableLocation);
+        // Verify we're registering a location with a valid table
+        try {
+            TableSnapshot tableSnapshot = transactionLogAccess.loadSnapshot(table.getSchemaTableName(), tableLocation, session);
+            transactionLogAccess.getMetadataEntry(tableSnapshot, session); // verify metadata exists
+        }
+        catch (IOException | RuntimeException e) {
+            throw new TrinoException(DELTA_LAKE_INVALID_TABLE, "Failed to access table location: " + tableLocation, e);
```

Comment:
Nit: This could double-wrap TrinoExceptions. Does it matter?

Reply:
good question. not intentional. i don't know, so will prevent this.